From 54ac563e431f3388845821b5618d222ac6e4c4c1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 2 Sep 2018 19:26:59 +0200
Subject: [PATCH] Various fixes related to gtk3 port

- Don't disable scrollbars, otherwise the window cannot be resized.
- Fix the alignment size of the intro text to make it visible.
- Set the default size of main window.
- Remove unimplemented Help Contents menu item.
---
 data/buoh-ui.xml        |  1 -
 src/buoh-view-comic.c   | 33 ---------------------------------
 src/buoh-view-message.c |  2 +-
 src/buoh-window.c       |  4 +---
 4 files changed, 2 insertions(+), 38 deletions(-)

diff --git a/data/buoh-ui.xml b/data/buoh-ui.xml
index 6531447..439b887 100644
--- a/data/buoh-ui.xml
+++ b/data/buoh-ui.xml
@@ -28,7 +28,6 @@
       <menuitem action="GoLast" name="GoLastMenu"/>
     </menu>
     <menu action="Help" name="HelpMenu">
-      <menuitem action="HelpContents" name="HelpContentsMenu"/>
       <menuitem action="HelpAbout" name="HelpAboutMenu"/>
     </menu>
   </menubar>
diff --git a/src/buoh-view-comic.c b/src/buoh-view-comic.c
index bd4338d..bad2fdf 100644
--- a/src/buoh-view-comic.c
+++ b/src/buoh-view-comic.c
@@ -103,7 +103,6 @@ static void     buoh_view_comic_prepare_load          (BuohViewComic    *c_view)
 static void     buoh_view_comic_load_finished         (BuohViewComic    *c_view,
                                                        gpointer          gdata);
 static void     buoh_view_comic_load                  (BuohViewComic    *c_view);
-static void     buoh_view_comic_update_scrollbar_policy (BuohViewComic  *c_view);
 static gdouble  buoh_view_comic_get_scale_for_width   (BuohViewComic    *c_view,
                                                        gint              width);
 static gdouble  buoh_view_comic_get_scale_for_height  (BuohViewComic    *c_view,
@@ -253,7 +252,6 @@ buoh_view_comic_set_property (GObject      *object,
                 break;
         case PROP_ZOOM_MODE:
                 c_view->priv->zoom_mode = g_value_get_enum (value);
-                buoh_view_comic_update_scrollbar_policy (c_view);
 
                 break;
         case PROP_SCALE:
@@ -573,8 +571,6 @@ buoh_view_comic_prepare_load (BuohViewComic *c_view)
         gtk_adjustment_set_value (hadjustment, 0.0);
         gtk_adjustment_set_value (vadjustment, 0.0);
 
-        buoh_view_comic_update_scrollbar_policy (c_view);
-
         if (gtk_widget_get_realized (GTK_WIDGET (c_view))) {
                 gdk_window_set_cursor (gtk_widget_get_window (GTK_WIDGET (c_view)),
                                        NULL);
@@ -788,35 +784,6 @@ buoh_view_comic_load (BuohViewComic *c_view)
         }
 }
 
-static void
-buoh_view_comic_update_scrollbar_policy (BuohViewComic *c_view)
-{
-        GtkWidget *swindow;
-
-        swindow = gtk_widget_get_parent (GTK_WIDGET (c_view));
-        if (!GTK_IS_SCROLLED_WINDOW (swindow)) {
-                return;
-        }
-
-        switch (c_view->priv->zoom_mode) {
-        case VIEW_ZOOM_FREE:
-                gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (swindow),
-                                                GTK_POLICY_AUTOMATIC,
-                                                GTK_POLICY_AUTOMATIC);
-                break;
-        case VIEW_ZOOM_FIT_WIDTH:
-                gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (swindow),
-                                                GTK_POLICY_NEVER,
-                                                GTK_POLICY_AUTOMATIC);
-                break;
-        case VIEW_ZOOM_BEST_FIT:
-                gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (swindow),
-                                                GTK_POLICY_NEVER,
-                                                GTK_POLICY_NEVER);
-                break;
-        }
-}
-
 static gdouble
 buoh_view_comic_get_scale_for_width (BuohViewComic *c_view,
                                      gint           width)
diff --git a/src/buoh-view-message.c b/src/buoh-view-message.c
index 69f9c5b..f82fd45 100644
--- a/src/buoh-view-message.c
+++ b/src/buoh-view-message.c
@@ -81,7 +81,7 @@ buoh_view_message_init (BuohViewMessage *m_view)
                             TRUE, TRUE, 0);
         gtk_widget_show (vbox);
 
-        align = gtk_alignment_new (0.5, 0.5, 0.0, 0.0);
+        align = gtk_alignment_new (0.5, 0.5, 1.0, 1.0);
         gtk_container_add (GTK_CONTAINER (align), hbox);
         gtk_widget_show (hbox);
 
diff --git a/src/buoh-window.c b/src/buoh-window.c
index 4859f07..038ed57 100644
--- a/src/buoh-window.c
+++ b/src/buoh-window.c
@@ -209,9 +209,6 @@ static const GtkActionEntry menu_entries[] = {
           G_CALLBACK (buoh_window_cmd_go_last) },
 
         /* Help menu */
-        { "HelpContents", GTK_STOCK_HELP, N_("_Contents"), "F1",
-          N_("Display help for the Buoh online comic reader"),
-          G_CALLBACK (buoh_window_cmd_help_contents) },
         { "HelpAbout", GTK_STOCK_ABOUT, N_("_About"), NULL,
           N_("Display credits for the Buoh online comic reader creators"),
           G_CALLBACK (buoh_window_cmd_help_about) }
@@ -255,6 +252,7 @@ buoh_window_init (BuohWindow *buoh_window)
 
         gtk_window_set_title (GTK_WINDOW (buoh_window), "Buoh");
         gtk_window_set_icon_name (GTK_WINDOW (buoh_window), "buoh");
+	gtk_window_set_default_size (GTK_WINDOW (buoh_window), 800, 400);
 
         buoh_window->priv->properties = NULL;
         buoh_window->priv->add_dialog = NULL;
-- 
2.18.0

